views/listTests.ejs

```diff
@@ -17,14 +41,25 @@
             <div class="panel-heading"><div style="width:100%">Les tests pour la campagne "<%= testSuite.name %>" réalisés
                 le <%= formattedTimestamp %></div>
 
-                <%if(!mine){%>
-                <button class="btn btn-default" onclick="window.location = '/listTestsConnected'">Voir seulement mes tests (login GitHub requis)</button>
-                <%} else {%>
-                <button class="btn btn-default" onclick="window.location = '/listTests'">Voir tous les tests</button>
-                <%}%>
+                <div class="row">
+                    <div class="col-sm-6">
+                        <%if(!mine){%>
+                        <button class="btn btn-default" onclick="filterMyTests()">Voir seulement mes tests (login GitHub requis)</button>
+                        <%} else {%>
+                        <button class="btn btn-default" onclick="filterAllTests()">Voir tous les tests</button>
+                        <%}%>
+                    </div>
+
+                    <div class="col-sm-6">
+                        <button class="btn btn-default pull-right" style="margin-right:25px;" onclick="filterByTestName()">Filtrer par nom de test</button>
+                        <input type="text" class="pull-right" name="testNameFilter" id="testNameFilter" placeholder="nom ou partie du nom d'un test"/>
+                        <button class="btn btn-default pull-right" style="position:absolute;top:0;right:0;width:35px;" onclick="resetTestNameFilter()">X</button>
```

Comment:
Use a `<form>` rather than JavaScript handlers if in the end you modify `window.location`.

> Using JS would make sense with the History API, but as progressive enhancement.